design-document.md

```diff
@@ -115,6 +293,50 @@ collude and break soundness. Is this a contingency we need to address? There are
 techniques in [BBC+19] that account for this; we need to figure out if they're
 practical.]]
 
+### Future work and possible extensions
+
+In this section we discuss attacks that are not considered in the above threat
+model, and suggest mitigations that could be incorporated into implementations
+of this protocol or future revisions of this specfication.
+
+#### Client authentication
+
+Attackers can impersonate Prio clients and submit large amounts of valid but
+false data in order to spoil aggregations. Deployments could require that
+clients present a trusted identity before they may contribute data. For example,
+by requiring data submissions to be signed with a key trusted by aggregators. A
+variety of techniques providing different degrees of assurance could be employed
+to establish trust in clients, ranging from whitebox cryptography to public key
+infrastructures. However some deployments may opt to accept the risk of false
+data to avoid having to figure out how to distribute trusted identities to
+clients.
```

Comment:
Another authentication mechanism: The leader might just require the user to login beforehand.

I'm curious how whitebox crypto might be used here?

Reply:
The notion with whitebox is that you could use it to distribute a shared secret to all "legitimate" instances of your client, and so having a message HMACed with a whiteboxed key gives you some kind of assurance on the provenance of the message. But thinking about it further, I've never been a big fan of whitebox, and mentioning it here isn't particularly helpful to explaining Prio, so I'll just strike it.